### PR TITLE
CI fixes

### PR DESCRIPTION
**Pull request by andrew-kaufman**
### Issues:
CryptoAlg-345

### Description of changes: 
Re-enables unit tests that were failing in CodeBuild.

### Testing:
CI tests are all calling `run_posix_tests.sh` and passing.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: patches/series [hunk 1]
```diff
@@ -4,3 +4,4 @@
 0004-disable-urandom-test-docker.patch
 0005-fix-ssl-test-runner-output.patch
 0006-fix-gcc4-compile-errors.patch
+0007-fix-abi-test-compilation.patch
```
Comment (bryce-shang) [inline patches/series:7]: `main` has 007 patch. Need to change this to 008 or 009.

File: patches/0007-fix-abi-test-compilation.patch [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+===================================================================
+--- aws-lc.orig/third_party/boringssl/crypto/test/abi_test.cc
++++ aws-lc/third_party/boringssl/crypto/test/abi_test.cc
+@@ -208,7 +208,8 @@ template <typename... Args>
+     WriteFile(stderr_handle, buf, strlen(buf), &unused, nullptr);
+   }
+ #else
+-  write(STDERR_FILENO, buf, strlen(buf));
++  OPENSSL_UNUSED ssize_t unused_ret =
++          write(STDERR_FILENO, buf, strlen(buf));
```
Comment (bryce-shang) [inline patches/0007-fix-abi-test-compilation.patch:37]: Is this to fix gcc4 related compile warning?

File: tests/coding_guidelines/c99_gcc_test.sh [hunk 3]
```diff
@@ -1,29 +1,20 @@
-#!/bin/bash -ex
-# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-# SPDX-License-Identifier: Apache-2.0
+#!/bin/bash
 
 # This test is aimed to run with GCC 6 and above on x86-64 architectures.
```
Comment (bryce-shang) [inline tests/coding_guidelines/c99_gcc_test.sh:3]: The copyright header is needed based on open source policy.

**Comment by andrew-kaufman** [comment]
Closing in favor of #40

**Pull request closed without merging**